File: wit.js
const axios = require("axios");
require("dotenv").config();

const accessToken = process.env.WIT_AI_ACCESS_TOKEN;

// Función para enviar un mensaje a Wit.ai y obtener la intención
async function obtenerIntencion(mensaje) {
    const url = "https://api.wit.ai/message";

    try {
        const response = await axios.get(url, {
            headers: { Authorization: `Bearer ${accessToken}` },
            params: { q: mensaje }
        });

        console.log("Respuesta de Wit.ai:", response.data); // Depuración

        // Verificar si hay intenciones detectadas
        const intents = response.data.intents;

        if (intents.length > 0) {
            return intents[0].name; // Retorna el nombre de la intención más relevante
        }

        return "por_defecto"; // Si no detecta intención, usa la respuesta por defecto
    } catch (error) {
        console.error("Error al conectar con Wit.ai:", error);
        return "por_defecto";
    }
}

module.exports = { obtenerIntencion };

File: chatbotPY/src/wit.py
import os
import requests
from dotenv import load_dotenv

load_dotenv()

ACCESS_TOKEN = os.getenv("ACCESS_TOKEN")

# Función para enviar un mensaje a Wit.ai y obtener la intención
def obtener_intencion(mensaje):
    url = "https://api.wit.ai/message"
    
    try:
        response = requests.get(url, 
                                headers={"Authorization": f"Bearer {ACCESS_TOKEN}"},
                                params={"q": mensaje})
        
        response_data = response.json()
        print("Respuesta de Wit.ai:", response_data)  # Depuración completa
        
        # Verificar si hay intenciones detectadas
        intents = response_data.get("intents", [])
        if intents:
            print(f"Intención detectada: {intents[0]['name']}")  # Imprimir la intención detectada
            return intents[0]["name"]  # Retorna la intención
        
        return "por_defecto"  # Si no detecta intención, usa la respuesta por defecto
    except Exception as e:
        print("Error al conectar con Wit.ai:", e)
        return "por_defecto"

